ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -765,6 +780,29 @@ module Winbind = struct
       hostname
 end
 
+let build_netbios_name ~config_params =
+  let key = "netbios-name" in
+  match List.assoc_opt key config_params with
+  | Some name ->
+      if String.length name > max_netbios_name_length then
+        raise
+          (Auth_service_error
+             ( E_GENERIC
+             , Printf.sprintf "%s cannot longger than %d chars" key
+                 max_netbios_name_length ))
+      else
+        name
+  | None ->
+      Winbind.build_netbios_name ()
+
+let build_dns_hostname_option ~config_params =
```

Comment:
What does the "option" in the name signal here? This function does not return an option.